### PR TITLE
feat: add isnull function

### PR DESCRIPTION
**Pull request by KKould**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Implement the `isnull` function to find remainders

- added a unit test
- updated the sqlness test

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link
issue: https://github.com/GreptimeTeam/greptimedb/issues/3355

### CODE REVIEW

File: tests/cases/standalone/common/function/expression.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+SELECT ISNULL(1);
+
+SELECT ISNULL(-1);
+
+SELECT ISNULL(1.0);
+
+SELECT ISNULL(true);
+
+SELECT ISNULL('string');
```
Comment (waynexia) [inline tests/cases/standalone/common/function/expression.sql:11]: Please add a tailing empty line

File: tests/cases/standalone/common/function/expression.sql
Comment (waynexia) [inline tests/cases/standalone/common/function/expression.sql:1]: What would happen if apply this function to a column?
```sql
select isnull(a) from t;
```

File: src/common/function/src/scalars/expression/is_null.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+use datatypes::data_type::ConcreteDataType;
+use datatypes::prelude::VectorRef;
+use datatypes::vectors::Helper;
+use snafu::{ensure, ResultExt};
+
+use crate::function::{Function, FunctionContext};
+
+const NAME: &str = "isnull";
+
+/// The function to find remainders
```
Comment (waynexia) [inline src/common/function/src/scalars/expression/is_null.rs:33]:
```suggestion
/// The function to check whether an expression is NULL
```

File: tests/cases/standalone/common/function/expression.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE t(a INTEGER, ts timestamp time index);
+
+INSERT INTO t VALUES (1, 1);
+
+INSERT INTO t VALUES (null, 2);
+
+INSERT INTO t VALUES (3, 3);
```
Comment (waynexia) [inline tests/cases/standalone/common/function/expression.sql:7]: nit-picking: merge them into one `INSERT` clause

**Review by waynexia — APPROVED** [review]
Others LGTM, appreciate it @KKould 🚀

File: tests/cases/standalone/common/function/expression.sql (hunk 3 above)
Comment (KKould) [reply tests/cases/standalone/common/function/expression.sql:7]: I'll adjust it now

**Review by waynexia — APPROVED** [review]

**Review by WenyXu — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your contribution!

**Comment by tisonkun** [comment]
@KKould out of curiosity, and since there is no linked good first issue, I wonder how you found this project :D

And welcome! You can browse other good first issue or start a discussion at [the Discussions forum](https://github.com/orgs/GreptimeTeam/discussions) for anything you are interested in this project.

**Comment by KKould** [comment]
> @KKould out of curiosity, and since there is no linked good first issue, I wonder how you found this project :D
> 
> And welcome! You can browse other good first issue or start a discussion at [the Discussions forum](https://github.com/orgs/GreptimeTeam/discussions) for anything you are interested in this project.

I was a database employee, I have been following Greptime for a long time, but I am more interested in things related to SQL queries. I have not encountered many related issues before.

**Pull request merged**